src/RandomForest.jl

```diff
@@ -0,0 +1,82 @@
+using StatsBase: mode, sample
+
+"""
+    RandomForest(data::Matrix{T}, labels::Vector{L}, max_depth::Int, min_samples_split::Int, number_of_trees::Int, subsample_percentage::Float64)
+    RandomForest(data::Matrix{T}, labels::Vector{L}, number_of_trees::Int, subsample_percentage::Float64)
+    RandomForest(data::Matrix{T}, labels::Vector{L})
+
+Represents a RandomForest.
+
+`trees` is the vector of ClassificationTree structures.
+`num_features` is the number of features to use when finding the best split. If -1, all the features are used.
+"""
+struct RandomForest{T, L}
+    trees::Vector{ClassificationTree{T, L}}
+    num_features::Int
+
+    function RandomForest(data::Matrix{T}, labels::Vector{L}, max_depth::Int, min_samples_split::Int ,number_of_trees::Int, subsample_percentage::Float64, num_features::Int) where {T, L}
```

Comment:
I would suggest moving the construcion and training of the trees to the fit function and including the data and labels as parameters in that function

Reply:
I agree. But to minimize merge conflicts. I would do the refactoring after we merged the regression and forest feature together :)